pkg/proxy/winkernel/hcnutils.go

```diff
@@ -346,13 +376,157 @@ func (hns hns) getLoadBalancer(endpoints []endpointsInfo, flags loadBalancerFlag
 	return lbInfo, err
 }
 
-func (hns hns) deleteLoadBalancer(hnsID string) error {
-	lb, err := hcn.GetLoadBalancerByID(hnsID)
+func (hns hcnutils) deleteLoadBalancer(hnsID string) error {
+	lb, err := hns.hcninstance.GetLoadBalancerByID(hnsID)
 	if err != nil {
 		// Return silently
 		return nil
 	}
 
-	err = lb.Delete()
+	err = hns.hcninstance.DeleteLoadBalancer(lb)
 	return err
 }
+
+func (h *RealHCN) GetNetworkByName(networkName string) (*hcn.HostComputeNetwork, error) {
+	return hcn.GetNetworkByName(networkName)
+}
+
+func (h *RealHCN) ListEndpointsOfNetwork(networkId string) ([]hcn.HostComputeEndpoint, error) {
+	return hcn.ListEndpointsOfNetwork(networkId)
+}
+
+func (h *RealHCN) GetEndpointByID(endpointId string) (*hcn.HostComputeEndpoint, error) {
+	return hcn.GetEndpointByID(endpointId)
+}
+
+func (h *RealHCN) ListEndpoints() ([]hcn.HostComputeEndpoint, error) {
+	return hcn.ListEndpoints()
+}
+
+func (h *RealHCN) GetEndpointByName(endpointName string) (*hcn.HostComputeEndpoint, error) {
+	return hcn.GetEndpointByName(endpointName)
+}
+
+func (h *RealHCN) ListLoadBalancers() ([]hcn.HostComputeLoadBalancer, error) {
+	return hcn.ListLoadBalancers()
+}
+
+func (h *RealHCN) GetLoadBalancerByID(loadBalancerId string) (*hcn.HostComputeLoadBalancer, error) {
+	return hcn.GetLoadBalancerByID(loadBalancerId)
+}
+
+func (h *RealHCN) CreateEndpoint(endpoint *hcn.HostComputeEndpoint, network *hcn.HostComputeNetwork) (*hcn.HostComputeEndpoint, error) {
+	return network.CreateEndpoint(endpoint)
+}
+
+func (h *RealHCN) CreateRemoteEndpoint(endpoint *hcn.HostComputeEndpoint, network *hcn.HostComputeNetwork) (*hcn.HostComputeEndpoint, error) {
+	return network.CreateRemoteEndpoint(endpoint)
+}
+
+func (h *RealHCN) CreateLoadBalancer(loadbalancer *hcn.HostComputeLoadBalancer) (*hcn.HostComputeLoadBalancer, error) {
+	return loadbalancer.Create()
+}
+
+func (h *RealHCN) DeleteLoadBalancer(loadbalancer *hcn.HostComputeLoadBalancer) error {
+	return loadbalancer.Delete()
+}
+
+func (h *RealHCN) DeleteEndpoint(endpoint *hcn.HostComputeEndpoint) error {
+	return endpoint.Delete()
+}
+
+// Here are some helper methods for proxier.go
+
+func getNetworkName(hnsNetworkName string) (string, error) {
+	if len(hnsNetworkName) == 0 {
```

Comment:
nit: I think ``hnsNetworkName == ""`` would be a more common approach. At the very least, it's cleaner.